app/utils/map.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useRef } from "react";
+
+/**
+ * Resize observer hook that can be used to trigger resize() when the
+ * map container size changes
+ * 
+ * @example
+ * const mapContainerRef = useResizeObserver<HTMLDivElement>(() => {
+    mapRef.current?.resize();
+  });
+ */
+export function useResizeObserver<T extends HTMLElement>(
+  callback: () => void,
+  debounceDelay: number = 50
+) {
+  const containerRef = useRef<T | null>(null);
+
+  useEffect(() => {
+    const observer = new ResizeObserver(() => {
+      // This timeout has a debouncing effect that prevents
+      // the map from flashing on the screen during sidebar
+      // animation. A more roboust implementation would clear the
+      // previous timeout interval but the performance hit
+      // seems neglibile
```

Comment:
couple of typos snuck in here

`roboust` → `robust` 
`neglibile` → `negligible`

Reply:
thank you!